### PR TITLE
Fix extensions: FE, MO, shared library and template extension op

### PR DESCRIPTION
**Pull request by slyalin**
Fixes for:

1. When loading a new OV operation with OPENVINO_FRAMEWORK_MAP macro from a separately compiled library, FE extensions generated by OPENVINO_FRAMEWORK_MAP are not added to FEs. It affects both core.add_extension and FE.add_extension, and both C++ and Python.
2. When using MO with --extensions that takes a separately compiled library, OPENVINO_FRAMEWORK_MAP extensions are not detected as well.
3. Setting shape for output tensor in template extension `evaluate` method leads to failure when this op runs in CPU plugin.

*No tests*. Need help from MO/ONNX/Paddle teams.

### CODE REVIEW

File: src/core/tests/frontend/onnx/CMakeLists.txt [hunk 1]
```diff
@@ -18,12 +18,13 @@ install(TARGETS ${TARGET_NAME}
         COMPONENT tests
         EXCLUDE_FROM_ALL)
 
-set(TEST_ONNX_MODELS_DIRNAME ${TEST_MODEL_ZOO}/src/core/models/onnx)
+set(TEST_ONNX_MODELS_DIRNAME ${TEST_MODEL_ZOO}/core/models/onnx)
```
Comment (ilyachur) [inline src/core/tests/frontend/onnx/CMakeLists.txt:21]: @tomdol Do we have such tests in CI? From my point of view path to models was incorrect...

Comment (tomdol) [reply src/core/tests/frontend/onnx/CMakeLists.txt:21]: This binary is not executed in CI (yet). There's a problem with this binary described in 80026 and this task is currently in progress.

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**